nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateRecord.java

```diff
@@ -0,0 +1,649 @@
+                    recordSchema = generateRecordSchema(fields, nullable);
+                    usingSchema = false;
+                }
+                try {
+                    final RecordSchema writeSchema = writerFactory.getSchema(attributes, recordSchema);
+                    try (final RecordSetWriter writer = writerFactory.createWriter(getLogger(), writeSchema, out, attributes)) {
+                        writer.beginRecordSet();
+
+                        Record record;
+                        List<RecordField> writeFieldNames = writeSchema.getFields();
+                        Map<String, Object> recordEntries = new HashMap<>();
+                        for (int i = 0; i < numRecords; i++) {
+                            for (RecordField writeRecordField : writeFieldNames) {
+                                final String writeFieldName = writeRecordField.getFieldName();
+                                final Object writeFieldValue;
+                                if (usingSchema) {
+                                    writeFieldValue = generateValueFromRecordField(writeRecordField, faker, nullPercentage);
+                                } else {
+                                    final boolean nullValue;
+                                    if (!context.getProperty(GenerateRecord.NULLABLE_FIELDS).asBoolean() || nullPercentage == 0) {
+                                        nullValue = false;
+                                    } else {
+                                        nullValue = (faker.number().numberBetween(0, 100) <= nullPercentage);
+                                    }
+                                    if (nullValue) {
+                                        writeFieldValue = null;
+                                    } else {
+                                        final String propertyValue = context.getProperty(writeFieldName).getValue();
+                                        writeFieldValue = getFakeData(propertyValue, faker);
+                                    }
+                                }
+
+                                recordEntries.put(writeFieldName, writeFieldValue);
+                            }
+                            record = new MapRecord(recordSchema, recordEntries);
+                            writer.write(record);
+                        }
+
+                        final WriteResult writeResult = writer.finishRecordSet();
+                        attributes.put("record.count", String.valueOf(writeResult.getRecordCount()));
+                        attributes.put(CoreAttributes.MIME_TYPE.key(), writer.getMimeType());
+                        attributes.putAll(writeResult.getAttributes());
+                        recordCount.set(writeResult.getRecordCount());
+                    }
+                } catch (final SchemaNotFoundException e) {
+                    throw new ProcessException(e.getLocalizedMessage(), e);
```

Comment:
The `localizedMessage` will be present in the stack trace, so it would be better to provide a specific message:
```suggestion
                    throw new ProcessException("Schema not found while writing records", e);
```